replica/distributed_loader.cc

```diff
@@ -649,33 +653,54 @@ future<> distributed_loader::init_non_system_keyspaces(distributed<replica::data
             }
         }).get();
 
-        std::vector<future<>> futures;
-
-        // treat "dirs" as immutable to avoid modifying it while still in 
-        // a range-iteration. Also to simplify the "finally"
-        for (auto i = dirs.begin(); i != dirs.end();) {
-            auto& ks_name = i->first;
-            auto e = dirs.equal_range(ks_name).second;
-            auto j = i++;
-            // might have more than one dir for a keyspace iff data_file_directories is > 1 and
-            // somehow someone placed sstables in more than one of them for a given ks. (import?) 
-            futures.emplace_back(parallel_for_each(j, e, [&](const std::pair<sstring, sstring>& p) {
-                auto& datadir = p.second;
-                return distributed_loader::populate_keyspace(db, datadir, ks_name);
-            }).finally([&] {
-                return db.invoke_on_all([ks_name] (replica::database& db) {
-                    // can be false if running test environment
-                    // or ks_name was just a borked directory not representing
-                    // a keyspace in schema tables.
-                    if (db.has_keyspace(ks_name)) {
-                        db.find_keyspace(ks_name).mark_as_populated();
-                    }
-                    return make_ready_future<>();
-                });
-            }));
-        }
+        bool prio_only = true;
+
+        for (;;) {
```

Comment:
How about: `for (auto prio_only : { true, false })`? I dislike loops that hide their termination condition. The termination condition explains why we're looping in the first place.

Reply:
Sure.